Dockerfile

```diff
@@ -1,23 +1,50 @@
 FROM python:3.12-slim AS base
 
-RUN apt-get update && apt-get install -y \
-    build-essential \
-    libpq-dev \
-    ffmpeg \
-    && rm -rf /var/lib/apt/lists/*
+ARG UID=1000
+ARG GID=1000
 
-ENV PYTHONUNBUFFERED 1
+RUN apt-get update && apt-get install -y --no-install-recommends \
+      ffmpeg \
+    && apt-get clean all \
+    && rm -rvf /var/lib/apt/* /var/cache/apt/* \
+    && find / -type d -name "__pycache__" -exec rm -rv {} + 2>/dev/null || true
+
+ENV VIRTUAL_ENV="/app_venv"
+ENV PATH="${VIRTUAL_ENV}/bin:${PATH}"
+
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+
+RUN \
+    groupadd -fg $GID app \
+    && useradd -m --shell /bin/false -r -u $UID -g app app
+
+RUN python3 -m venv "${VIRTUAL_ENV}"
 
 WORKDIR /app
-COPY . /app/
 
-RUN pip install --no-cache-dir -r requirements/base.txt
+FROM base AS app_venv_production
+
+COPY requirements/base.txt /tmp/requirements/base.txt
+RUN pip install --no-cache-dir --no-compile -r /tmp/requirements/base.txt \
+    && pip install --no-cache-dir --no-compile gunicorn
 
 FROM base AS app
 
-COPY entrypoint.sh /entrypoint.sh
-RUN chmod +x /entrypoint.sh
+COPY --from=app_venv_production ${VIRTUAL_ENV} ${VIRTUAL_ENV}
 
+COPY . /app/
+
+RUN chmod +x entrypoint.sh
```

Comment:
The `app` user doesnot seem to have permissions to execute `entrypoint.sh`. I guess the command to make entrypoint.sh executable is run with root user.
<img width="1521" alt="Screenshot 2025-02-12 at 8 35 16 PM" src="https://github.com/user-attachments/assets/be221aba-85f0-4a27-9695-a5bd8b1e1eea" />

Reply:
I have found the problem. In addition to the `RUN` command being wrong, the `/app/` directory is being mounted in the container in docker-compose file, presumably for local development. The `entrypoint.sh` script in this repository is not set to executable. I have set the executable attribute to the script and tested that it works fine.